models/build_run_results.go

```diff
@@ -0,0 +1,96 @@
+package models
+
+import (
+	"time"
+
+	"github.com/bitrise-io/bitrise/v2/exitcode"
+)
+
+type BuildRunResultsModel struct {
+	WorkflowID           string                `json:"workflow_id" yaml:"workflow_id"`
+	EventName            string                `json:"event_name" yaml:"event_name"`
+	ProjectType          string                `json:"project_type" yaml:"project_type"`
+	StartTime            time.Time             `json:"start_time" yaml:"start_time"`
+	StepmanUpdates       map[string]int        `json:"stepman_updates" yaml:"stepman_updates"`
+	SuccessSteps         []StepRunResultsModel `json:"success_steps" yaml:"success_steps"`
+	FailedSteps          []StepRunResultsModel `json:"failed_steps" yaml:"failed_steps"`
+	FailedSkippableSteps []StepRunResultsModel `json:"failed_skippable_steps" yaml:"failed_skippable_steps"`
+	SkippedSteps         []StepRunResultsModel `json:"skipped_steps" yaml:"skipped_steps"`
+}
+
+func NewBuildRunResultsModel(workflowID string, start time.Time, projectType string) BuildRunResultsModel {
+	return BuildRunResultsModel{
+		WorkflowID:     workflowID,
+		StartTime:      start,
+		StepmanUpdates: map[string]int{},
+		ProjectType:    projectType,
+	}
+}
+
+func (buildRes BuildRunResultsModel) IsStepLibUpdated(stepLib string) bool {
+	return (buildRes.StepmanUpdates[stepLib] > 0)
+}
+
+func (buildRes BuildRunResultsModel) IsBuildFailed() bool {
+	return len(buildRes.FailedSteps) > 0
+}
+
+func (buildRes BuildRunResultsModel) ExitCode() int {
+	if !buildRes.IsBuildFailed() {
+		return 0
```

Comment:
You could also use `exitcode.CLISuccess` in this case.

(Only later saw that these are just copy pasted existing lines.)